lib/turn.rb

```diff
@@ -7,29 +7,29 @@ def initialize(player1, player2)
   end
 
   def type
-    if zero_index_rank(@player1) != zero_index_rank(@player2)
+    if player1.deck.rank_of_card_at(0) != player2.deck.rank_of_card_at(0)
       :basic
-    elsif zero_index_rank(@player1) == zero_index_rank(@player2) &&
-      two_index_rank(@player1) != two_index_rank(@player2)
-      :war
-    else
+    elsif zero_index_rank(player1) == zero_index_rank(player2) &&
+      two_index_rank(player1) == two_index_rank(player2)
       :mutually_assured_destruction
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)
+      :war
     end
   end
 
   def winner
     if type == :basic
-      if zero_index_rank(@player1) > zero_index_rank(@player2)
```

Comment:
Removed a lot of the instance variable "@" after a meeting with my mentor. She explained that we really only need to call the "@" on an instance that we are trying to change or mutate. Since we're just referencing player1 and player2 here, we were able to remove the "@" and still have the program run smoothly.